### PR TITLE
create intmin/intmax functions in pkg/util/integer

### PR DESCRIPTION
**Pull request by mqliang**
ref https://github.com/kubernetes/kubernetes/issues/20863

### CODE REVIEW

**Comment by mqliang** [comment]
Need rework when https://github.com/kubernetes/kubernetes/pull/20368 and https://github.com/kubernetes/kubernetes/pull/20696 are merged.

**Comment by mqliang** [comment]
@bgrant0607 @janetkuo @kargakis

File: pkg/controller/deployment/deployment_controller.go [hunk 1]
```diff
@@ -39,6 +38,7 @@ import (
 	"k8s.io/kubernetes/pkg/util"
 	deploymentutil "k8s.io/kubernetes/pkg/util/deployment"
 	utilerrors "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/interger"
```
Comment (janetkuo) [inline pkg/controller/deployment/deployment_controller.go]: typo: should be "integer"

File: pkg/util/interger/interger.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package interger
+
+func Max(a, b int) int {
```
Comment (janetkuo) [inline pkg/util/interger/interger.go]: Use `int64` type instead?

Comment (mqliang) [inline pkg/util/interger/interger.go]: Or `IntMax` instead, and add `Int64Max`, so we can eliminate type cast?

Comment (bgrant0607) [inline pkg/util/interger/interger.go]: I like IntMax and Int64Max.

**Comment by bgrant0607** [comment]
Thanks!

**Comment by bgrant0607** [comment]
LGTM. Please rebase.

**Comment by bgrant0607** [comment]
LGTM

**Comment by bgrant0607** [comment]
@k8s-bot test this #21045

**Comment by mqliang** [comment]
@bgrant0607 Rebased. Could you kindly please reapply LGTM.

**Comment by bgrant0607** [comment]
LGTM

**Pull request merged**